Ballerina-Training/Ballerina-Forum-Application/back-end/bal-forum/main.bal

```diff
@@ -0,0 +1,222 @@
+import ballerina/http;
+import ballerina/sql;
+import ballerina/task;
+import ballerina/time;
+import ballerina/uuid;
+
+configurable int PORT = 4000;
+listener http:Listener forumServerEP = new (PORT);
+
+@http:ServiceConfig {
+    cors: {
+        allowOrigins: ["*"]
+    }
+}
+service /api on forumServerEP {
+    resource function post users(NewUser newUser) returns Created|Conflict|error {
+        User|error user = forumDbClient->queryRow(`
+            SELECT id, name, email FROM users WHERE name = ${newUser.name}
+        `);
+        if user is User {
+            Conflict userExists = {
+                body: {
+                    error_message: "User already exists"
+                }
+            };
+            return userExists;
+        }
+
+        string id = uuid:createType1AsString();
+        _ = check forumDbClient->execute(`
+            INSERT INTO users VALUES (${id}, ${newUser.name}, ${newUser.email}, ${newUser.password}, "[]")
+        `);
+
+        _ = start sendEmailNotification(newUser);
+
+        Created userCreated = {
+            body: {
+                message: "User created successfully!"
+            }
+        };
+        return userCreated;
+    }
+
+    resource function get users/[string id]() returns User|NotFound|error {
+        UserDb|error userDb = forumDbClient->queryRow(`
+            SELECT name, email, id, subscribtions FROM users WHERE id = ${id}
+        `);
+        if userDb is UserDb {
+            User user = check transformUserFromDatabase(userDb);
+            return user;
```

Comment:
Would also combine here, since the check is unnecessary.

Reply:
+1

Reply:
Fixed via [1eae7ab](https://github.com/ballerina-guides/training-content/pull/12/commits/1eae7abbc8d1d6bdd3a677b4fac348875ee53fb7)